src/sagemaker/utils.py

```diff
@@ -1362,3 +1363,47 @@ def update_nested_dictionary_with_values_from_config(
     )
 
     return inferred_config_dict
+
+
+def stringify_object(obj: Any) -> str:
+    """Returns string representation of object, returning only non-None fields."""
+    non_none_atts = {key: value for key, value in obj.__dict__.items() if value is not None}
+    return f"{type(obj).__name__}: {str(non_none_atts)}"
+
+
+def volume_size_supported(instance_type: str) -> bool:
+    """Returns True if SageMaker allows volume_size to be used for the instance type.
+
+    Raises:
+        ValueError: If the instance type is improperly formatted.
+    """
+
+    try:
+
+        # local mode does not support volume size
+        if instance_type.startswith("local"):
```

Comment:
will need to add another check `or is_pipeline_variable(instance_type)`, otherwise, this breaks pipeline cx.